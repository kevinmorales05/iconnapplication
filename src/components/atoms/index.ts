export * from './AccountItem';
export * from './ActionButton';
export * from './AddressItems';
export * from './AnnounceItem';
export * from './Avatar';
export * from './BackButton';
export * from './CardAction';
export * from './Checkbox';
export * from './ConnectionItem';
export * from './Container';
export * from './CustomModal';
export * from './CustomText';
export * from './DatePicker';
export * from './EmptyCardsCard';
export * from './EmptyQRCard';
export * from './InfoCard';
export * from './Input';
export * from './PaymentCard';
export * from './SafeArea';
export * from './Section';
export * from './Select';
export * from './ServiceCard';
export * from './Touchable';
