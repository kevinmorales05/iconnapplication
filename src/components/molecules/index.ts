export * from './Accordion';
export * from './AddressCard';
export * from './BasketCounter';
export * from './Button';
export * from './CardBilling';
export * from './CardDivided';
export * from './CardHorizontal';
export * from './CardImage';
export * from './CheckButton';
export * from './Code';
export * from './CountryCodeSelect';
export * from './EcommerceHeader';
export * from './FavoriteButton';
export * from './ImagesCarusel';
export * from './ImageZoom';
export * from './ListSwipeableItem';
export * from './ModalCard';
export * from './NavigationMenuItem';
export * from './OrderCard';
export * from './PriceWithDiscount';
export * from './QuantityProduct';
export * from './RadioButton';
export * from './Rating';
export * from './SearchBar';
export * from './SearchItem';
export * from './TaxInfoCard';
export * from './TextContainer';
export * from './TouchableText';
