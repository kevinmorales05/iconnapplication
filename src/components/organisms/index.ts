export * from './AboutEmail';
export * from './Alert';
export * from './Loading';
export * from './OtherInputMethods';
export * from './InvoicingHelper';
export * from './ResendInvoice';
export * from './AddressModalSelection';
export * from './ShippingDropdown';
