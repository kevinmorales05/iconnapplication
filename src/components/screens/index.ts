export * from './Billing';
export * from './InvoiceGenerated';
