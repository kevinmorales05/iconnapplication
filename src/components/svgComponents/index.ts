export * from './SearchLoupeDeleteSvg';
export * from './StarSvg';
export * from './HeartSvg';
export * from './HeartSvgOutline';
export * from './BasketSvg';
export * from './CloseSvg';
