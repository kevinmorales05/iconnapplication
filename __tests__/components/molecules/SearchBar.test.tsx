import React from 'react';
import { SearchBar } from 'components';
import renderer from 'react-test-renderer';

const onPressDoNothing = () => {};

jest.mock('react-native-device-info', () => {
  return {
    getVersion: () => 4,
    hasNotch: () => true,
    getModel: () => 'iPhone 8 Plus'
  };
});
jest.mock('@react-native-google-signin/google-signin', () => {});
jest.mock('@react-native-async-storage/async-storage', () => {});
jest.mock('react-native-permissions', () => {});
jest.mock('@react-native-community/netinfo', () => {});
jest.mock('react-native-blob-util', () => {});
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');
jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest.fn().mockImplementation((config, reducers) => reducers)
  };
});

describe('Testing on <SearchBar />', () => {
  test('should match with the snapshot as input text', () => {
    const snap = renderer
      .create(
        <SearchBar
          onChangeTextSearch={onPressDoNothing}
          onPressSearch={onPressDoNothing}
          isButton={false}
          onEndWriting={onPressDoNothing}
          textSearch="something"
          placeHolderText="placeholder"
        />
      )
      .toJSON();

    expect(snap).toMatchSnapshot();
  });
  test('should match with the snapshot as button', () => {
    const snap = renderer.create(<SearchBar onChangeTextSearch={onPressDoNothing} onPressSearch={onPressDoNothing} isButton={true} />).toJSON();

    expect(snap).toMatchSnapshot();
  });
});
