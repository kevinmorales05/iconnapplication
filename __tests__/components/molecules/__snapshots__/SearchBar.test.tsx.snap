// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <SearchBar /> should match with the snapshot as button 1`] = `
<View
  collapsable={false}
  style={
    Array [
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
      ],
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#dadadb",
        "borderRadius": 15.714285714285714,
        "borderWidth": 1.5714285714285714,
        "flexDirection": "row",
        "height": 62.857142857142854,
        "width": "100%",
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Array [
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
        ],
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#747476",
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    collapsable={false}
    style={
      Array [
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
        ],
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": undefined,
          "left": undefined,
          "right": undefined,
          "top": undefined,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 0,
          "height": undefined,
          "marginBottom": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
          "opacity": 1,
          "width": undefined,
          "zIndex": 2,
        }
      }
    >
      <View
        pointerEvents="none"
      >
        <Text
          ellipsizeMode="tail"
          style={
            Array [
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 15,
                },
                Object {
                  "color": "#747476",
                },
              ],
              Object {
                "alignSelf": undefined,
                "letterSpacing": undefined,
                "lineHeight": undefined,
                "textAlign": undefined,
                "textDecorationLine": "none",
                "textTransform": undefined,
              },
            ]
          }
        >
          ¿Qué estás buscando?
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Testing on <SearchBar /> should match with the snapshot as input text 1`] = `
<View
  collapsable={false}
  style={
    Array [
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
        Object {},
      ],
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#dadadb",
        "borderRadius": 15.714285714285714,
        "borderWidth": 1.5714285714285714,
        "flexDirection": "row",
        "height": 62.857142857142854,
        "width": "100%",
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Array [
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
        ],
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#747476",
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "Feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    collapsable={false}
    style={
      Array [
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
          Object {},
        ],
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <TextInput
      multiline={false}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      placeholder="¿Qué estás buscando?"
      placeholderTextColor="#747476"
      style={
        Object {
          "color": "#000000",
        }
      }
      value=""
    />
  </View>
</View>
`;
